android/examples/Example_Application/app/src/main/java/com/ironcorelabs/example_application/MainActivity.java

```diff
@@ -0,0 +1,140 @@
+package com.ironcorelabs.example_application;
+
+import android.app.Dialog;
+import android.content.DialogInterface;
+import android.os.Bundle;
+import android.text.Editable;
+import android.util.Log;
+import android.view.View;
+import android.widget.EditText;
+import android.widget.TextView;
+
+import androidx.annotation.NonNull;
+import androidx.appcompat.app.AlertDialog;
+import androidx.appcompat.app.AppCompatActivity;
+import androidx.fragment.app.DialogFragment;
+import androidx.recyclerview.widget.LinearLayoutManager;
+import androidx.recyclerview.widget.RecyclerView;
+
+import com.ironcorelabs.sdk.DeviceContext;
+import com.ironcorelabs.sdk.DocumentDecryptResult;
+import com.ironcorelabs.sdk.DocumentEncryptOpts;
+import com.ironcorelabs.sdk.DocumentEncryptResult;
+import com.ironcorelabs.sdk.DocumentName;
+import com.ironcorelabs.sdk.GroupId;
+import com.ironcorelabs.sdk.IronOxide;
+import com.ironcorelabs.sdk.IronOxideConfig;
+import com.ironcorelabs.sdk.UserId;
+
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.Objects;
+
+public class MainActivity extends AppCompatActivity implements MyRecyclerViewAdapter.ItemClickListener {
+    private EditText encryptText;
+    private EditText encryptId;
+    private TextView decryptView;
+    private IronOxide sdk;
+    private final ArrayList<DocumentEncryptResult> encryptedList = new ArrayList<>();
+    private MyRecyclerViewAdapter adapter;
+
+    private IronOxide initializeSdk() {
+        System.loadLibrary("ironoxide_android");
+        InputStream in = getResources().openRawResource(R.raw.device_context);
+        final java.util.Scanner s = new java.util.Scanner(in).useDelimiter("\\A");
```

Comment:
I think some explanation of why this is a good way to read in a `DeviceContext` would be nice, assuming this _is_ a good way...